bindings/java/src/org/sleuthkit/datamodel/BlackboardArtifact.java

```diff
@@ -170,12 +170,25 @@ public String getDisplayName() {
 	 */
 	public String getShortDescription() throws TskCoreException {
 		BlackboardAttribute attr = null;
-		if (artifactTypeId == ARTIFACT_TYPE.TSK_WEB_BOOKMARK.getTypeID()) {
-			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_URL));
+		if (artifactTypeId == ARTIFACT_TYPE.TSK_WEB_BOOKMARK.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_WEB_COOKIE.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_WEB_DOWNLOAD.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_WEB_HISTORY.getTypeID()) {
+			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DOMAIN));
 		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID()) {
 			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_KEYWORD_PREVIEW));
-		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_CALLLOG.getTypeID()) {
-			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM));
+		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_DEVICE_ATTACHED.getTypeID()) {
+			attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DEVICE_ID));
+		} else if (artifactTypeId == ARTIFACT_TYPE.TSK_CONTACT.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_MESSAGE.getTypeID()
+				|| artifactTypeId == ARTIFACT_TYPE.TSK_CALLLOG.getTypeID()) {
+			if (null != getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER))) {
+				attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER));
+			} else if (null != getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM))) {
+				attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM));
+			} else if (null != getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_TO))) {
+				attr = getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_TO));
+			}
```

Comment:
There exists custom module code (see JIRA for details) that uses a different set of phone number attributes for a TSK_CONTACT artifact: TSK_PHONE_NUMBER_HOME,  TSK_PHONE_NUMBER_OFFICE, TSK_PHONE_NUMBER_MOBILE, or TSK_PHONE_NUMBER. The same code potentially also, or only, attaches email address attributes: TSK_EMAIL_HOME, TSK_EMAIL_OFFICE, or TSK_EMAIL.  And in org.sleuthkit.autopsy.modules.iOS.ContactAnalyzer, there will be either TSK_PHONE_NUMBER or TSK_EMAIL, but not both, as is possible for the custom module. 

The custom code also creates TSK_MESSAGE artifacts that may be either phone messages (TSK_PHONE_NUMBER_TO) or email messages (TSK_EMAIL_TO). 

While the central repository/correlation engine (CR/CE) code in EmailArtifactUtil.getTypeFromBlackboardArtifact only deals with  TSK_PHONE_NUMBER_FROM, TSK_PHONE_NUMBER_TO, and TSK_PHONE_NUMBER, I think this method should work for general use, not just the CR/CE use case. @bcarrier, as the designer of the short description feature and as a SME,  should phone nuumbers take precedence over email addresses? Should one type of phone number take precedence over another type of phone number? I am thinking specificfally of TSK_CONTACT artifacts that may contani mulitple phone numbers and email addresses for a single individual.